client/transactions/list/test/index.tsx

```diff
@@ -415,16 +426,28 @@ describe( 'Transactions list', () => {
 			} );
 		} );
 
-		afterEach( () => {
-			jest.resetAllMocks();
-		} );
+		test( 'should request confirmation when the download button is clicked', () => {
+			mockUseTransactionsSummary.mockReturnValue( {
+				transactionsSummary: {
+					count: 2000,
+				},
+				isLoading: false,
+			} );
+
+			window.confirm = jest.fn();
 
-		afterAll( () => {
-			jest.restoreAllMocks();
+			const { getByRole } = render( <TransactionsList /> );
+			getByRole( 'button', { name: 'Download' } ).click();
+
+			expect( window.confirm ).toHaveBeenCalledTimes( 1 );
+			expect( window.confirm ).toHaveBeenCalledWith(
```

Comment:
May we mock the windown.confirm value, and verify that the code will trigger this API request too? 

https://github.com/Automattic/woocommerce-payments/blob/bebc1d686c8287e5408f671d28e687341b0e6ab7/client/transactions/list/index.tsx#L435-L436

Reply:
IMO, if we can mock `apiFetch` and verify the parameters with which it has been called, that would be great.

Reply:
Sure thing! I made these changes in https://github.com/Automattic/woocommerce-payments/pull/3477/commits/aa7647f2a055b23da9341ba3413e8d4b059ae13f